x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/adapter_types.ts

```diff
@@ -42,6 +39,13 @@ export interface GetMonitorStatesResult {
   prevPagePagination: string | null;
 }
 
+export interface GetSnapshotCountParams {
+  dateRangeStart: string;
+  dateRangeEnd: string;
+  filters?: string | null;
+  statusFilter?: string | null;
+}
+
```

Comment:
it's generally makes sense to move all the interfaces to the top of the file, can you move this close to MonitorStatesParams? 
And also may be use MonitorStatesParams for snapshot as well, since pagination is optional. Once you do that i guess it will also makes sense to have a bit more generic name Just QueryParams, WDYT?

Reply:
We already have that in the first Maps PR, when we merge one we can overwrite this for the other and yes we can just use a generic interface for components that only require these four basic params I agree!